src/engine/StripColumns.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2025 The QLever Authors, in particular:
+//
+// 2025 Johannes Kalmbach <kalmbach@cs.uni-freiburg.de>, UFR/QL
+//
+// UFR = University of Freiburg, Chair of Algorithms and Data Structures
+// QL =  QLeverize AG
+
+// You may not use this file except in compliance with the Apache 2.0 License,
+// which can be found in the `LICENSE` file at the root of the QLever project.
+
+#include "engine/StripColumns.h"
+
+#include "engine/QueryExecutionTree.h"
+
+// _____________________________________________________________________________
+StripColumns::StripColumns(QueryExecutionContext* ctx,
+                           std::shared_ptr<QueryExecutionTree> child,
+                           const std::set<Variable>& keepVariables)
```

Comment:
Given that the rest of the code is hidden somewhere in #2239, is there a specific reason why a set is chosen here? Why couldn't you just use a `std::vector` and just eliminate duplicates on the fly, or forbid them using a `CONTRACT_CHECK`?

Reply:
The code is not performance critical, and we want the variables to be
1. unique (clearer semantically)
2. ordered (for better cache resusage).